### PR TITLE
feat: adds a new flag --no-server to start and use surge TUI without the HTTP server

### PR DESCRIPTION
**Pull request by junaid2005p**
<h3>Greptile Summary</h3>

This PR adds a `--no-server` flag to the `surge` root command that starts the TUI without binding the embedded HTTP API server, making the instance fully local and unreachable by CLI subcommands or browser extensions. The implementation is clean and well-tested.

- `maybeStartRootHTTPServer` wraps the existing `startRootHTTPServer`, returning `port=0` and a no-op cleanup when `--no-server` is set, and surfaces an error when `--port` is combined with `--no-server`.
- The TUI header now renders \"Local mode\" in gray (suppressing the green dot and server address) whenever `ServerPort == 0 && !IsRemote`, with regression tests for both wide and compact layouts.

<h3>Confidence Score: 5/5</h3>

Safe to merge — the change is well-scoped, the new code paths are covered by tests, and the existing server startup path is unchanged.

The flag is wired end-to-end: option parsing → server skip → TUI display. Conflict detection for --port --no-server is explicit and tested. No existing behavior is altered when the flag is absent.

No files require special attention.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| cmd/root.go | Adds noServer/portSet fields to rootRunOptions, readRootRunOptions, and introduces maybeStartRootHTTPServer which short-circuits when --no-server is set and returns an error when combined with --port. |
| cmd/cmd_test.go | Adds four new tests: flag registration check, readRootRunOptions capturing --no-server, portSet tracking, maybeStartRootHTTPServer skipping the port file, and error on --port+--no-server conflict. |
| internal/tui/view_dashboard_header.go | Updates the header to show Local mode in gray (instead of Serving at 127.0.0.1:0) and suppresses the green status dot when ServerPort == 0 && !IsRemote. |
| internal/tui/view_test.go | Adds two regression tests covering wide and compact header rendering in local/no-server mode, verifying the address and dot are suppressed. |
| README.md | Documents the --no-server flag with an example invocation and a note that CLI subcommands won't reach the instance. |
| docs/USAGE.md | Adds --no-server to the command reference table with a brief description of its effect. |

</details>

<details open><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `cmd/root.go`, line 467-484 ([link](https://github.com/surgedm/surge/blob/6d9bca8bdacbd7e30e690e47e3c99785aa73129c/cmd/root.go#L467-L484)) 

   <a href="#">P2</a> **No tests for the new `--no-server` flag**

   The flag introduces two new code paths — skipping `startRootHTTPServer` and the `resolveAPIConnection` guard — but no tests cover them. Edge cases worth testing include: the TUI starting correctly when `port=0` is passed, and the error message returned when a caller needs a server (`requireServer=true`) while `noServer=true`.

   **Rule Used:** What: All code changes must include tests for edge... ([source](https://app.greptile.com/surge-org-3/-/custom-context?memory=2b22782d-3452-4d55-b059-e631b2540ce8))
   
   <details><summary>Prompt To Fix With AI</summary>
   
   `````markdown
   This is a comment left during a code review.
   Path: cmd/root.go
   Line: 467-484
   
   Comment:
   **No tests for the new `--no-server` flag**
   
   The flag introduces two new code paths — skipping `startRootHTTPServer` and the `resolveAPIConnection` guard — but no tests cover them. Edge cases worth testing include: the TUI starting correctly when `port=0` is passed, and the error message returned when a caller needs a server (`requireServer=true`) while `noServer=true`.
   
   **Rule Used:** What: All code changes must include tests for edge... ([source](https://app.greptile.com/surge-org-3/-/custom-context?memory=2b22782d-3452-4d55-b059-e631b2540ce8))
   
   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>
</details>

<sub>Reviews (6): Last reviewed commit: ["fix: addressed port issue marked by grep..."](https://github.com/surgedm/surge/commit/8b52aed2500b952c525269e0c3e2e6e23fc0028d) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=34794655)</sub>

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -463,7 +464,17 @@ var rootCmd = &cobra.Command{
...
 
 		opts := readRootRunOptions(cmd)
-		port, cleanup, err := startRootHTTPServer(opts)
+		noServerFromFlag, _ := cmd.Flags().GetBool("no-server")
+		if noServerFromFlag {
+			noServer = true
+		}
+		var port int
+		var cleanup func()
+		if noServer {
```
[the remark this answers is not in the document]
Comment (junaid2005p) [reply cmd/root.go]: globalHost and globalToken are used throughout the codebase but no-server is used locally so it does not make sense to bind it to a struct

File: cmd/root.go [hunk 2]
```diff
@@ -595,6 +606,7 @@ func init() {
 	rootCmd.Flags().StringP("output", "o", "", "Output directory (defaults to current working directory)")
 	rootCmd.Flags().Bool("no-resume", false, "Do not auto-resume paused downloads on startup")
 	rootCmd.Flags().Bool("exit-when-done", false, "Exit when all downloads complete")
+	rootCmd.Flags().Bool("no-server", false, "Do not start the HTTP API server (CLI subcommands will not work)")
```
[the remark this answers is not in the document]
Comment (junaid2005p) [reply cmd/root.go:613]: globalHost and globalToken are used throughout the codebase but no-server is used locally so it does not make sense to bind it to a struc

File: cmd/root.go [hunk 3]
```diff
@@ -463,7 +464,17 @@ var rootCmd = &cobra.Command{
...
 		opts := readRootRunOptions(cmd)
-		port, cleanup, err := startRootHTTPServer(opts)
+		noServerFromFlag, _ := cmd.Flags().GetBool("no-server")
+		if noServerFromFlag {
+			noServer = true
+		}
+		var port int
+		var cleanup func()
+		if noServer {
+			port, cleanup, err = 0, func() {}, nil
```
[the remark this answers is not in the document]
Comment (junaid2005p) [reply cmd/root.go]: Check again the fix is already there? if not tell me
